textattack/loggers/visdom_logger.py

```diff
@@ -16,9 +17,22 @@ def __init__(self, env='main', port=8097, hostname='localhost'):
         if not port_is_open(port, hostname=hostname):
             raise socket.error(f'Visdom not running on {hostname}:{port}')
         self.vis = Visdom(port=port, server=hostname, env=env)
+        self.vis_args = {'port': port, 'server': hostname, 'env': env}
         self.windows = {}
         self.sample_rows = []
 
+    def __getstate__(self):
+        # Temporarily save socket handle b/c we can't copy it
+        tmp = self.vis
+        self.vis = None
+        state = copy.deepcopy(self.__dict__)
+        self.vis = tmp
+        return state
+
```

Comment:
I see this pattern here and `FileLogger` and was trying to think of a better way. Here's another option, using a dictionary comprehension:

`state = {i: d[i] for i in self.__dict__ if i !='vis'}`

not saying this is necessarily better, it's up to you whether you'd want to change the two `__getstate__` methods to look like this. It's probably a little less easy to comprehend this way but a decent amount more concise. Up to you.

Reply:
I agree that this is better.